models/Asset/Image.php

```diff
@@ -403,7 +403,11 @@ public function getWidth()
     {
         $dimensions = $this->getDimensions();
 
-        return $dimensions['width'];
+        if (isset($dimensions)) {
```

Comment:
```suggestion
        if ($dimensions) {
```
As `null` is returned in case the dimensions cannot be read, a simple `if` should be sufficient.